### ISSUE TITLE
no answer, no exception

### ISSUE THREAD

**Issue by adeptofvoltron** [mpociot/botman#232]
I just made like in tutorial
```php
use Mpociot\BotMan\BotManFactory;
use Mpociot\BotMan\BotMan;
use Mpociot\BotMan\Storages\Drivers\FileStorage;

$token = 'MY-TOKEN';
$config =  [
    'facebook_token' => $token,
    'facebook_app_secret' => 'MY-SECRET',

];

// $storage = new FileStorage(__DIR__);
// create an instance
$botman = BotManFactory::create($config);
$botman->verifyServices($token);

$botman->hears('foo', function ($bot) {
    $bot->typesAndWaits(2);
    $bot->reply('Hello World');
});
// give the bot something to listen for.
$botman->hears('.*', function (BotMan $bot) {
    $bot->reply('Hello yourself.');
});

// start listening
$botman->listen();
```

I see in logs...that facebook made request to my address. but no answer appear. Also didn't get anythng in error log.

**Comment by mpociot** [comment]
Please make sure that you are the administrator of the Facebook page.
Either that, or the token might be wrong.

Did the Facebook verification process work?

**Comment by adeptofvoltron** [comment]
yes. it work.
I'm the owner of the page. Also owner of facebook Application. 
The App is not published. but it should work for person connected with App.

**Comment by christophrumpel** [comment]
Did you use a correct `facebook_app_secret` and `token` ?

**Comment by adeptofvoltron** [comment]
So, on botman factory I logged:
`$request->getContent()` and it gaves me
```js
{"object":"page","entry":[{"id":"121648734848223","time":1487614765582,"messaging":[{"sender":{"id":"1428632740512537"},"recipient":{"id":"121648734848223"},"timestamp":1487614765416,"message":{"mid":"mid.1487614765416:2c1bd34019","seq":178222,"text":"foo"}}]}]}
```
and in Facebook Driver in the end getMessages i dumpet to logs $message var

```php
array (
  0 =>
  Mpociot\BotMan\Message::__set_state(array(
     'message' => '',
     'user' => '',
     'channel' => '',
     'payload' => NULL,
     'extras' =>
    array (
    ),
  )),
)
```

so message content is empty

**Comment by takid1412** [comment]
I just found that `validateSignature` function of `FacebookDriver` always return `false` and that make Message empty. I must return true for quick fix but less secure
```
protected function validateSignature()
    {
    	return true;
        return hash_equals($this->signature, 'sha1='.hash_hmac('sha1', $this->content, $this->config->get('facebook_app_secret')));
    }
```

**Comment by christophrumpel** [comment]
@takid1412 and you're sure that the `facebook_app_secret` is set correctly in your `services` config file and in the `.env` file?

```php
'facebook_app_secret' => env('FACEBOOK_APP_SECRET')
```

```php
FACEBOOK_APP_SECRET=your-app-secret
```

**Comment by takid1412** [comment]
@christophrumpel Yeah, I'm sure about that. 
I dumped config variable just before creating Botman object and every element is match. Only problem at that function, other works as expected.

**Comment by christophrumpel** [comment]
Is the whole code online somewhere? You are using it in a plain php project right? I'd like to test it on my machine too if possible.

**Comment by takid1412** [comment]
Today i un-comment that line of code and suddenly it works. I'm really confused, maybe facebook has changed something at that days. So thanks for your support.

**Comment by adeptofvoltron** [comment]
@mpociot I will make some tests during the weekend. If the problem will be the same I will reopen(open new issue).
good luck!

**Comment by kenkit** [comment]
I'm facing the same issue with telegram. No message response but my message does get through the hear.
But I don't get a reply back.
Also using the first example with php7 from php5

**Comment by abr4xas** [comment]
Hi, I'm facing the same issue. 

I'm the owner of the fb page. Also owner of facebook app.

How to set botman facebook bot to production??

**Comment by christophrumpel** [comment]
What is exactly your problem @abr4xas ? Please describe it in more detail.

**Comment by abr4xas** [comment]
@christophrumpel I dont have any response only works with me, the owner of the fb page but using another user account  I dont have any response.

**Comment by christophrumpel** [comment]
Ok, so then please check:

* is your FB app still in sandbox mode?
* is you FB page already published?
* have you done the FB app review process?

**Comment by abr4xas** [comment]
* is your FB app still in sandbox mode?
   * No
* is you FB page already published?
   * Yes
* have you done the FB app review process?
   * Yes, i need `pages_messaging` permission

**Comment by christophrumpel** [comment]
Have you tried it with different (not-admin) user? Last you could also try to a new fb app or new fb page. This is annoying but I also had occasions where this helped.

**Comment by salmanyaqoobin** [comment]
Hi,
I am experiencing the same issue, i created FB page, created FB app, connect with my server.
Currently every thing is working fine for me only on FB messenger bcz im administrator for FB app and FB page. but when some else try to chat with my page messenger he is not getting any response from messenger.
I read above conversation regarding to page_messages permission, i didnot find this sort of permission under my FB app. Can you please guide me, so the chat bot will be active for other users.
My FB page link: https://www.facebook.com/International-Nomads-Bots-249022525792866/
Thanks

**Comment by christophrumpel** [comment]
So if you want other users (not admin and testers) to use your chatbot you need the `pages_messaging` permission on your app. You can find it under your FB app settings (not the page) -> messenger settings. You will need to send your app in review before and FB will decide if you get the permission. Only after that, you should use your chatbot publicly.


<img width="1228" alt="screen shot 2018-11-21 at 7 57 32 am" src="https://user-images.githubusercontent.com/1394539/48823994-269a6d00-ed63-11e8-83f0-b48a15727480.png">
